pkgs/development/tools/build-managers/gradle/default.nix

```diff
@@ -147,4 +147,43 @@ rec {
     sha256 = "16iqh4bn7ndch51h2lgkdqyyhnd91fdfjx55fa3z3scdacl0491y";
     defaultJava = jdk11;
   };
+
+  wrapGradle = {
+      lib, callPackage, mitm-cache, substituteAll, symlinkJoin, concatTextFile, makeSetupHook,
+    }:
+    gradle-unwrapped:
+    lib.makeOverridable (args:
+    let
+      gradle = gradle-unwrapped.override args;
+      build-support = ../../../../build-support/gradle;
+    in symlinkJoin {
+      name = "gradle-${gradle.version}";
+
+      paths = [
+        (makeSetupHook { name = "gradle-setup-hook"; } (concatTextFile {
+          name = "setup-hook.sh";
+          files = [
+            (mitm-cache.setupHook)
+            (substituteAll {
+              src = /${build-support}/setup-hook.sh;
+              # jdk used for keytool
+              inherit jdk;
```

Comment:
Instead of using `jdk` here, have you considered adding `passthru.jdk` to "unwrapped" Gradle, and using it here? This would avoiding pulling in yet another JDK in some cases. If you did and decided against it, I'd appreciate if you could explain your reasoning. Thanks!